packages/config/src/utils/extensions/auto-install-extensions.ts

```diff
@@ -0,0 +1,138 @@
+import { promises as fs } from 'fs'
+
+import { NetlifyAPI } from 'netlify'
+
+import { JIGSAW_URL } from './constants.js'
+import {
+  getExtension,
+  getInstalledExtensionsForSite,
+  installExtension,
+  InstallExtensionResult,
+  getAccount,
+} from './utils.js'
+
+export async function handleAutoInstallExtensions(opts: $TSFixMe, api?: NetlifyAPI) {
+  if (!opts.accountId || !opts.token || !api?.accessToken) {
+    // skip installing extensions if not logged in
+    return
+  }
+
+  const installableExtensions = await getExtensionsToInstall(opts, api)
+  if (installableExtensions.length === 0) {
+    return
+  }
+
+  const accountId = opts.accountId
+  const account = await getAccount(api, { accountId })
+
+  const results: InstallExtensionResult[] = await Promise.all(
+    installableExtensions.map((installableExt) => {
+      if (!installableExt.hostSiteUrl) {
+        return {
+          slug: installableExt.slug,
+          error: {
+            code: 'FAILED_TO_FETCH_EXTENSION',
+            message: 'Failed to fetch extension host site url',
+          },
+        }
+      }
+
+      console.log(
+        `Installing extension "${installableExt.name ?? installableExt.slug}" on team "${
+          account.name
+        }" required by package(s): "${installableExt.meta.packages.join('",')}"`,
+      )
+
+      return installExtension({
+        accountId: accountId,
+        netlifyToken: opts.token,
+        slug: installableExt.slug,
+        hostSiteUrl: installableExt.hostSiteUrl,
+      })
+    }),
+  )
+
+  results.forEach(({ error, slug }) => {
+    if (error) {
+      console.warn(`Failed to install "${slug}" extension on team "${account.name}": ${error.message}`)
+      return
+    }
+    const ext = installableExtensions.find((ext) => ext.slug === slug)
+    console.log(`Installed${ext?.name ? ` "${ext.name}" ` : ` "${slug}" `}extension on team "${account.name}"`)
+  })
+}
+
+async function getExtensionsToInstall(opts: $TSFixMe, api: NetlifyAPI) {
+  if (!opts.accountId || !api.accessToken) {
+    // skip installing extensions if not logged in
+    return []
+  }
+
+  const accountId = opts.accountId
+  const siteId = opts.siteId
+
+  const getPackageJSON = async () => {
+    const packageJsonPath = `${opts.repositoryRoot}/package.json`
```

Comment:
This needs to change.

Reply:
If goal is to see what user already have installed to avoid autoinstalling it I would suggest using `require.resolve` instead of inspecting `package.json` contents or using already existing helpers (that use it) that we have for build plugins at least - like so https://github.com/netlify/build/blob/cffdcedfb01015c9d6bbf96c8f527ebad6a6d092/packages/build/src/plugins/resolve.js#L119-L124

(possibly those helpers would need to be moved around as they do exist in `netlify/build` and not `netlify/config`)

Reply:
😄 I switched to using require to load the package.json here: https://github.com/netlify/build/pull/6284/files#diff-440639a03b2eed8902e401f4354492ffabd4290581eff97455e7243c72bf904fR10-R13